src/main/java/moli/Storage.java

```diff
@@ -0,0 +1,93 @@
+package moli;
+
+import moli.task.Task;
+import moli.task.Todo;
+import moli.task.Deadline;
+import moli.task.Event;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private static final String FILE_PATH = "./data/moli.txt";
+
+    /**
+     * Saves the task list to a file.
+     *
+     * @param tasks The list of tasks to save.
+     */
+    public static void saveTasks(ArrayList<Task> tasks) {
+        try {
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs(); // Create directory if it doesn't exist
+            BufferedWriter writer = new BufferedWriter(new FileWriter(file));
+
+            for (int i = 0; i < tasks.size(); i++) {
+                writer.write(tasks.get(i).toFileFormat() + System.lineSeparator());
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads the task list from a file.
+     *
+     * @return The list of tasks.
+     */
+    public static ArrayList<Task> loadTasks() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File file = new File(FILE_PATH);
+
+        if (!file.exists()) {
+            return tasks; // Return empty list if file doesn't exist
+        }
```

Comment:
Good job keeping the [happy path prominent](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent).